pkg/authorization/registry/subjectaccessreview/rest.go

```diff
@@ -34,12 +35,25 @@ func (r *REST) Create(ctx kapi.Context, obj runtime.Object) (runtime.Object, err
 		return nil, fmt.Errorf("not a subjectAccessReview: %#v", obj)
 	}
 
-	namespace := kapi.NamespaceValue(ctx)
-	user := &user.DefaultInfo{
-		Name:   subjectAccessReview.User,
-		Groups: subjectAccessReview.Groups,
+	var userToCheck user.Info
+	if (len(subjectAccessReview.User) == 0) && (len(subjectAccessReview.Groups) == 0) {
+		// if no user or group was specified, use the info from the context
+		exists := false
+		userToCheck, exists = kapi.UserFrom(ctx)
+		if !exists {
+			return nil, errors.New("user missing from context")
+		}
+
+	} else {
+		userToCheck = &user.DefaultInfo{
+			Name:   subjectAccessReview.User,
+			Groups: subjectAccessReview.Groups.List(),
+		}
+
 	}
-	requestContext := kapi.WithUser(kapi.WithNamespace(kapi.NewDefaultContext(), namespace), user)
+
+	namespace := kapi.NamespaceValue(ctx)
+	requestContext := kapi.WithUser(kapi.WithNamespace(kapi.NewDefaultContext(), namespace), userToCheck)
```

Comment:
@smarterclayton I thought we always wanted to branch contexts with changed values off the one we already had, rather than calling `NewDefaultContext()`, otherwise we lose other context values like tracing info. Should this be doing `requestContext := kapi.WithUser(ctx, userToCheck)` instead?